src/game/savegame/savegame_bson.c

```diff
@@ -609,6 +604,12 @@ static bool SaveGame_BSON_LoadFx(struct json_array_s *fx_arr)
             fx->frame_number = frame_number;
             fx->counter = counter;
             fx->shade = shade;
+
+            // Map recreated FX to parent item if needed.
+            if (parent_item != -1 && object_number == O_FLAME
+                && g_Config.enable_enhanced_saves) {
```

Comment:
`&& g_Config.enable_enhanced_saves` is redundant - this is already tested at the beginning of the function.